src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -1838,9 +1838,16 @@ public MinifiedRunspaceDetails(RunspaceDetails eventArgs)
         /// <param name="e">details of the execution status change</param>
         private void PowerShellContext_ExecutionStatusChangedAsync(object sender, ExecutionStatusChangedEventArgs e)
         {
-            _languageServer?.SendNotification(
-                "powerShell/executionStatusChanged",
-                e);
+            // The cancelling of the prompt (PSReadLine) causes an ExecutionStatus.Aborted to be sent after every
+            // actual execution (ExecutionStatus.Running) on the pipeline. We ignore that event since it's counterintuitive to
+            // the goal of this method which is to send updates when the pipeline is actually running something.
+            // In the event that we don't know if it was a ReadLine cancel, we default to sending the notification.
+            if (!e?.ExecutionOptions?.IsReadLine ?? true)
```

Comment:
It's really hard to tell what this is doing, maybe change to something like this:

```csharp
var options = e?.ExecutionOptions;
if (options == null || !options.IsReadLine)
```

Reply:
but I like question marks 😄

will change 👍

Reply:
I used your logic